ARM builds on Windows unsupported with Version 2.5.0

While arm builds on windows where fine with version 2.4.0 commit https://github.com/uclouvain/openjpeg/commit/1e6c925eb5f6b68097a9592ca4df4f24eb8b9e56 broke them 

## Expected behavior and actual behavior.

a successful build on ARM

## Steps to reproduce the problem.

Checkout https://github.com/microsoft/vcpkg/pull/24734/
run bootstrap.(sh|bat) to get vcpkg
`vcpkg install openjpeg:arm64-windows`

## Operating system

windows

## openjpeg version

2.5.0

Error:
```
FAILED: bin/openjp2.dll bin/openjp2.lib 
cmd.exe /C "cd . && D:\downloads\tools\cmake-3.22.2-windows\cmake-3.22.2-windows-i386\bin\cmake.exe -E vs_link_dll --intdir=src\lib\openjp2\CMakeFiles\openjp2.dir --rc=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\rc.exe --mt=C:\PROGRA~2\WI3CF2~1\10\bin\100190~1.0\x64\mt.exe --manifests  -- C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1431~1.311\bin\Hostx64\arm64\link.exe  src\lib\openjp2\CMakeFiles\openjp2.dir\thread.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\bio.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\cio.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\dwt.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\event.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\ht_dec.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\image.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\invert.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\j2k.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\jp2.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\mct.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\mqc.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\openjpeg.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\opj_clock.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\pi.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\t1.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\t2.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\tcd.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\tgt.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\function_list.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\opj_malloc.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\sparse_array.c.obj  /out:bin\openjp2.dll /implib:bin\openjp2.lib /pdb:bin\openjp2.pdb /dll /version:2.5 /machine:ARM64 /nologo    /debug /INCREMENTAL  kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib  && cd ."
LINK Pass 1: command "C:\PROGRA~1\MICROS~1\2022\ENTERP~1\VC\Tools\MSVC\1431~1.311\bin\Hostx64\arm64\link.exe src\lib\openjp2\CMakeFiles\openjp2.dir\thread.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\bio.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\cio.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\dwt.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\event.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\ht_dec.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\image.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\invert.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\j2k.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\jp2.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\mct.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\mqc.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\openjpeg.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\opj_clock.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\pi.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\t1.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\t2.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\tcd.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\tgt.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\function_list.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\opj_malloc.c.obj src\lib\openjp2\CMakeFiles\openjp2.dir\sparse_array.c.obj /out:bin\openjp2.dll /implib:bin\openjp2.lib /pdb:bin\openjp2.pdb /dll /version:2.5 /machine:ARM64 /nologo /debug /INCREMENTAL kernel32.lib user32.lib gdi32.lib winspool.lib shell32.lib ole32.lib oleaut32.lib uuid.lib comdlg32.lib advapi32.lib /MANIFEST /MANIFESTFILE:src\lib\openjp2\CMakeFiles\openjp2.dir/intermediate.manifest src\lib\openjp2\CMakeFiles\openjp2.dir/manifest.res" failed (exit code 1120) with the following output:
   Creating library bin\openjp2.lib and object bin\openjp2.exp
ht_dec.c.obj : error LNK2019: unresolved external symbol __popcnt referenced in function population_count
bin\openjp2.dll : fatal error LNK1120: 1 unresolved externals
```

Reason: the Code switches on the compiler and ignores the architecture. Please only use intrinsics if the architecture is also met.

--- a/src/lib/openjp2/ht_dec.c
+++ b/src/lib/openjp2/ht_dec.c
@@ -69,7 +69,7 @@ static OPJ_BOOL only_cleanup_pass_is_decoded = OPJ_FALSE;
 static INLINE
 OPJ_UINT32 population_count(OPJ_UINT32 val)
 {
-#ifdef OPJ_COMPILER_MSVC
+#if defined(OPJ_COMPILER_MSVC) && (defined(_M_IX86) || defined(_M_AMD64))
     return (OPJ_UINT32)__popcnt(val);
 #elif (defined OPJ_COMPILER_GNUC)
     return (OPJ_UINT32)__builtin_popcount(val);

